dashboard/app/models/contact_rollups_processed.rb

```diff
@@ -183,6 +207,49 @@ def self.extract_professional_learning_attended(contact_data)
     return uniq_courses.blank? ? {} : {professional_learning_attended: uniq_courses}
   end
 
+  def self.extract_roles(contact_data)
+    roles = []
+    # Contact is a teacher if they appears in any of the following tables
+    roles << 'Teacher' if
+      contact_data.dig('dashboard.users', 'user_id') ||
+      contact_data.key?('dashboard.pd_enrollments') ||
+      contact_data.key?('dashboard.pd_attendances') ||
+      contact_data.key?('dashboard.followers')
+
+    # Contact is a teacher if they submit a census survey as a teacher
+    submitter_roles = extract_field(contact_data, 'dashboard.census_submissions', 'submitter_role') || []
```

Comment:
do we need to do this check if we already know the user is a teacher?

Reply:
That's a good point. I will add a check to avoid redundant work.